### PR TITLE
Test Skeleton

### PR DESCRIPTION
**Pull request by mpatrou**
# Short description of the changes:
The code includes a skeleton for testing with folder arrangements, documentation and gitlfs data repository inclusion.

# Long description of the changes:
The changes for testing are:
- folder organization with README, regarding the contents of the directory
- garnet-data gitlfs submodule and instructions
- test with garnet-data and related marker/condition, in case it exists
- docs on testing process

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer
Run pytest locally
without and with garnet-data repository configured. Tests should be skipped or passed, depending on the case.

To configure garnet-data:

Open a terminal and go to garnet repository root folder. Run:

- git submodule add https://code.ornl.gov/sns-hfir-scse/infrastructure/test-data/garnet-data tests/data/garnet-data

To initialize the data-repo. Run:
- git submodule update --init --recursive

# References
 https://ornl-neutrons.atlassian.net/wiki/spaces/NDPD/pages/19103745/Using+git-lfs+for+test+data

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import pytest
+import os
+
+
+@pytest.fixture(scope="session")
+def has_datarepo():
```
Comment (ghost) [inline tests/conftest.py:7]: This isn't properly checking the data repo availability. The directory exists even before calling `git submodule init` and `git submodule update` so `has_datarepo` is returning true but the test using it is failing.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    """Fixture that returns True if the datarepo_dir is available"""
+    root_data = os.path.join(os.path.dirname(os.path.abspath(__file__)), "data", "garnet-data")
+    return os.path.exists(root_data)
```
Comment (ghost) [inline tests/conftest.py]: pytest fixtures can use other fixtures so you could have `datarepo_dir` and this fixture could be re-written as 
```python
@pytest.fixture(scope="session")
def has_datarepo(datarepo_dir):
    return os.path.exists(datarepo_dir)
```

File: tests/utility/test_garnet_data.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    assert os.path.exists(datarepo_dir)
+
+    # ensure we can access the test file
+    test_file = os.path.join(datarepo_dir, "test.txt")
+    assert os.path.exists(test_file)
+
+    # ensure we can read the contents of the file
+    test_data = open(test_file)
+    with open(test_file, "r") as f:
+        test_data = json.load(f)
```
Comment (ghost) [inline tests/utility/test_garnet_data.py:22]: Technically not in this repo but, there is an error in the Json data. `b` is missing a comma after so `json.load` throws `json.decoder.JSONDecodeError`
```json
{
"a":1,
"b":2
"c":3
}
```

File: pyproject.toml [hunk 4]
```diff
@@ -44,7 +44,7 @@ testpaths = ["tests"]
 python_files = ["test*.py"]
 norecursedirs = [".git", "tmp*", "_tmp*", "__pycache__", "*dataset*", "*data_set*"]
 markers = [
-    "mymarker: example markers goes here"
+    "datarepo: example markers goes here"
```
Comment (ghost) [inline pyproject.toml]: `example markers goes here` could maybe be replaced with something like `Tests that require the test-data submodule`

File: tests/tabs/home/views/test_welcome.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+from garnet import Garnet, __version__
+
+
+def test_welcome_label(qtbot):
+    """Test starting the app"""
+
+    # initialization
+    garnet = Garnet()
+    garnet.show()
+    qtbot.waitUntil(garnet.show, timeout=5000)
```
Comment (ghost) [inline tests/tabs/home/views/test_welcome.py]: [Recommended to register window with qtbot](https://pytest-qt.readthedocs.io/en/latest/reference.html#pytestqt.qtbot.QtBot.addWidget)  take it or leave it, just throwing it out there...

File: docs/getting_started.rst [hunk 6]
```diff
@@ -34,3 +34,110 @@ If it has been a while, once can update using
...
+    conda activate <garnet_environment>
+    cd /path/to/my/local/garnet/repo/
+    git fetch origin pull/<PULL_REQUEST_NUMBER>/head:pr<PULL_REQUEST_NUMBER>
+    git switch pr<PULL_REQUEST_NUMBER>
+    #run tests
+    python -m pytest <item_to_test>
+    #and/or start garnet
+    garnet
+
+in case there is specific Mantid build in another conda environment, garnet can be installed there:
```
Comment (ghost) [inline docs/getting_started.rst]: I'm not sure what this block is supposed to mean?

File: docs/getting_started.rst [hunk 7]
```diff
@@ -34,3 +34,110 @@ If it has been a while, once can update using
...
+- data/ : It contains the test data. Small data files can be included in this directory. Forl large data files, there is a subfolder garner-data
+ that is connected to a gitlfs data repo in gitlab as a submodule: https://code.ornl.gov/sns-hfir-scse/infrastructure/test-data/garnet-data
+- tabs/ : Tests are organized into folders for every page:
+    * home/
+    * order_param/
+    * peak_integr/
+    * recp_space/
+    * ub_peak_find/
+
+        In every page-tab directory the tests are furtherd arranged in MVP-based filesystem structure:
```
Comment (ghost) [inline docs/getting_started.rst:50]: Screenshot from 2024-02-20 21-01-59 Is this how you intended it to be displayed? Sphinx may read this file different but in Github this sort of looks unintended.

Comment (mpatrou) [reply docs/getting_started.rst:50]: nope. done

File: pyproject.toml (hunk 4 above)
Comment (mpatrou) [reply pyproject.toml]: done!

File: tests/conftest.py (hunk 1 above)
Comment (mpatrou) [reply tests/conftest.py:7]: changed the check condition.

File: tests/conftest.py (hunk 2 above)
Comment (mpatrou) [reply tests/conftest.py]: N/A

File: tests/utility/test_garnet_data.py (hunk 3 above)
Comment (mpatrou) [reply tests/utility/test_garnet_data.py:22]: done.

File: docs/getting_started.rst (hunk 6 above)
Comment (mpatrou) [reply docs/getting_started.rst]: done

File: tests/tabs/home/views/test_welcome.py (hunk 5 above)
Comment (mpatrou) [reply tests/tabs/home/views/test_welcome.py]: done

**Review by ghost — APPROVED** [review]
Looks great, don't worry about the failing test/pre-commit. I already fixed that in my PR

**Pull request merged**